src/web/mjs/connectors/MangaParkEN.mjs

```diff
@@ -10,43 +9,121 @@ export default class MangaParkEN extends Connector {
         super.label = 'MangaPark';
         this.tags = [ 'manga', 'multi-lingual' ];
         this.url = 'https://mangapark.net';
+        this.apiURL = `${this.url}/apo/`;
+
+        this.requestOptions.headers.set('x-origin', this.url);
+        this.requestOptions.headers.set('x-referer', `${this.url}/`);
         this.requestOptions.headers.set('x-cookie', 'set=h=1;');
     }
 
     async _getMangaFromURI(uri) {
         const request = new Request(uri, this.requestOptions);
-        const queryMangaTitle = /\/title\/\d+/.test(uri.pathname) ? 'main h3' : 'meta[property="og:title"]';
+        const queryMangaTitle = /\/title\/\d+/.test(uri.pathname) ? 'main h3 a' : 'meta[property="og:title"]';
         const data = await this.fetchDOM(request, queryMangaTitle);
-        return new Manga(this, uri.pathname.match(/\/(\d+)\/?/)[1], (data[0].textContent || data[0].content).trim());
+        const id = parseInt(uri.pathname.match(/\/(\d+)\/?/)[1]);
+        const title = (data[0].textContent || data[0].content).trim();
+        return new Manga(this, id, title);
     }
 
     async _getMangas() {
+        try {
+            const mangaList = await this._getMangasV3();
+            if (mangaList.length == 0) {
+                throw new Error('Got 0 mangas from site using v3 version! Using v5 version as a fallback!');
```

Comment:
Suggestion to call and return V5 directly instead using `try catch`, since catch will also be performed for other errors (out of memory, no network connection, ...)

Reply:
I would do that if in my environment I wasn't getting 404 errors when trying to receive manga/chapters via `v3` endpoint.

For example, using code below I can't fetch manga via `v5` endpoint because `this._getMangasV3()` method throws a 404 error:
```js
async _getMangas() {
  const mangaList = await this._getMangasV3();
  if (mangaList.length == 0) {
	  return await this._getMangasV5();
  }
  return mangaList;
}
```
Log:
![image](https://user-images.githubusercontent.com/70586473/224565913-54062d3f-dc04-43a1-a3df-6e23d130022f.png)

When using VPN `v3` endpoint works, but I think other people can get this problem too.